Sources/RInAppMessaging/CampaignDispatcher.swift

```diff
@@ -167,4 +180,68 @@ internal class CampaignDispatcher: CampaignDispatcherType, TaskSchedulable {
             completion(data)
         }.resume()
     }
+
+    func fetchImagesArray(from carousel: Carousel, completion: @escaping ([UIImage?]) -> Void) {
+        guard let imageDetails = carousel.images else {
+            completion([])
+            return
+        }
+
+        let filteredDetails = imageDetails
+            .sorted { $0.key < $1.key }
+            .prefix(5)
+            .map { $0 }
+
+        let dispatchGroup = DispatchGroup()
+        var images: [UIImage?] = Array(repeating: nil, count: filteredDetails.count)
+
+        for (index, detail) in filteredDetails.enumerated() {
+            guard let urlString = detail.value.imgUrl else {
+                images[index] = nil
+                continue
+            }
+            
+            dispatchGroup.enter()
+            fetchCarouselImage(for: urlString) { image in
+                images[index] = image
+                dispatchGroup.leave()
```

Comment:
DispatchGroup is not required here, it is useful when there are multiple asynchronous tasks.
Inside `fetchCarouselImage` return `completion(images)`